### ISSUE TITLE
[prefer-ngsrc] [prefer-template-literal] cannot be enabled together when use base64 image

### ISSUE THREAD

**Issue by Yiding-Aya** [angular-eslint/angular-eslint#2887]
**Description and reproduction of the issue**

The '@angular-eslint/template/prefer-ngsrc': 'error' and  '@angular-eslint/template/prefer-template-literal': 'error' cannot be used together when base64 image is used.

Fix `prefer-template-literal` would fail `prefer-ngsrc` and fix `prefer-ngsrc` would fail `prefer-template-literal`. 

Issues with configuration and rules should be easily reproducible on StackBlitz.

Please fork this StackBlitz and your updated link below:

https://stackblitz.com/edit/angular-eslint-repros-uudw1bpu?file=src%2Fmain.ts

**Versions**

| package                                  | version |
| ---------------------------------------- | ------- |
| `@angular-eslint` | `21.2.0` |
| `ESLint`                                 | `9.32.2` |
| `node`                                   | `24.11.0` |

```sh
# Please run `npx ng version` in your project and paste the full output here:
┌───────────────────────────────────┬───────────────────┬───────────────────┐
│ Package                           │ Installed Version │ Requested Version │
├───────────────────────────────────┼───────────────────┼───────────────────┤
│ @angular-devkit/core              │ 21.1.2            │ ^21.1.2           │
│ @angular-devkit/schematics        │ 21.1.2            │ ^21.1.2           │
│ @angular/animations               │ 21.1.2            │ ^21.1.2           │
│ @angular/build                    │ 21.1.2            │ ^21.1.2           │
│ @angular/cdk                      │ 21.1.2            │ ^21.1.2           │
│ @angular/cli                      │ 21.1.2            │ ^21.1.2           │
│ @angular/common                   │ 21.1.2            │ ^21.1.2           │
│ @angular/compiler                 │ 21.1.2            │ ^21.1.2           │
│ @angular/compiler-cli             │ 21.1.2            │ ^21.1.2           │
│ @angular/core                     │ 21.1.2            │ ^21.1.2           │
│ @angular/forms                    │ 21.1.2            │ ^21.1.2           │
│ @angular/google-maps              │ 21.1.2            │ ^21.1.2           │
│ @angular/language-service         │ 21.1.2            │ ^21.1.2           │
│ @angular/material                 │ 21.1.2            │ ^21.1.2           │
│ @angular/platform-browser         │ 21.1.2            │ ^21.1.2           │
│ @angular/platform-browser-dynamic │ 21.1.2            │ ^21.1.2           │
│ @angular/router                   │ 21.1.2            │ ^21.1.2           │
│ @schematics/angular               │ 21.1.2            │ ^21.1.2           │
│ rxjs                              │ 7.8.2             │ ^7.8.2            │
│ typescript                        │ 5.9.3             │ ~5.9.3            │
│ zone.js                           │ 0.16.0            │ ~0.16.0           │

```

- [ ] I have tried restarting my IDE and the issue persists.
- [ ] I have updated to the latest supported version of the packages and checked my `ng version` output per the instructions given here.

**Comment by reduckted** [comment]
This seems like a similar problem to #2906.
